src/main/java/com/solvd/movie/persistence/impl/SearchMovieRepositoryImpl.java

```diff
@@ -0,0 +1,70 @@
+package com.solvd.movie.persistence.impl;
+
+import com.solvd.movie.model.EsMovie;
+import com.solvd.movie.model.criteria.SearchCriteria;
+import com.solvd.movie.persistence.SearchMovieRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.elasticsearch.core.ReactiveElasticsearchOperations;
+import org.springframework.data.elasticsearch.core.SearchHit;
+import org.springframework.data.elasticsearch.core.mapping.IndexCoordinates;
+import org.springframework.data.elasticsearch.core.query.Criteria;
+import org.springframework.data.elasticsearch.core.query.CriteriaQuery;
+import org.springframework.data.elasticsearch.core.query.Query;
+import org.springframework.stereotype.Repository;
+import reactor.core.publisher.Flux;
+
+@Repository
+@RequiredArgsConstructor
+public class SearchMovieRepositoryImpl implements SearchMovieRepository {
+
+    private static final String INDEX = "movies";
+    private final ReactiveElasticsearchOperations operations;
+
+    @Override
+    public Flux<EsMovie> findAllByCriteria(final SearchCriteria searchCriteria,
+                                           final Pageable pageable) {
+        Criteria criteria = new Criteria();
+        if (searchCriteria.getName() != null) {
+            criteria.and(Criteria.where("name")
```

Comment:
@hizmailovich This logic is not scalable enough, simply because when you need to add another criteria/field to search you will modify this code, so make it out this algorithm has few components, that will be much reusable

```java
new CriteriaOf(search); // will denote search as List<Field> fields
// and add all of them to criteria according to each Fieald policy #apply()

interface Field {
void apply();
}
```